browser/android/brave_sync_worker.cc

```diff
@@ -61,9 +92,305 @@ static void JNI_BraveSyncWorker_MarkSyncV1WasEnabledAndMigrated(
+      // because something changed since the UI was displayed. A passphrase
+      // cannot be set in such circumstances.
+      passphrase_failed = true;
+    } else {
+      // OK, the user sent us a passphrase, but we don't have pending keys. So
+      // it either means that the pending keys were resolved somehow since the
+      // time the UI was displayed (re-encryption, pending passphrase change,
+      // etc) or the user wants to re-encrypt.
+      if (configuration.set_new_passphrase &&
+          !service->GetUserSettings()->IsUsingSecondaryPassphrase()) {
+        service->GetUserSettings()->SetEncryptionPassphrase(
+            configuration.passphrase);
+      }
+    }
+  }
+
+  if (passphrase_failed ||
+      service->GetUserSettings()->IsPassphraseRequiredForPreferredDataTypes()) {
+    VLOG(1) << __func__ << " setup passphrase failed";
+  }
+
+  if (configuration.encrypt_all)
+    ProfileMetrics::LogProfileSyncInfo(ProfileMetrics::SYNC_ENCRYPT);
+  if (!configuration.set_new_passphrase && !configuration.passphrase.empty())
+    ProfileMetrics::LogProfileSyncInfo(ProfileMetrics::SYNC_PASSPHRASE);
+}
+
+static void JNI_BraveSyncWorker_Init(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jcaller) {
+  new BraveSyncWorker(env, jcaller);
+}
+
+base::android::ScopedJavaLocalRef<jstring>
+JNI_BraveSyncWorker_GetSeedHexFromWords(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jcaller,
+    const base::android::JavaParamRef<jstring>& seed_words) {
+  std::string str_seed_words =
+      base::android::ConvertJavaStringToUTF8(seed_words);
+  DCHECK(!str_seed_words.empty());
+
+  std::string sync_code_hex;
+  std::vector<uint8_t> bytes;
+  if (brave_sync::crypto::PassphraseToBytes32(str_seed_words, &bytes)) {
+    DCHECK_EQ(bytes.size(), SEED_BYTES_COUNT);
+    sync_code_hex = base::HexEncode(&bytes.at(0), bytes.size());
+  } else {
+    VLOG(1) << __func__ << " PassphraseToBytes32 failed for " << str_seed_words;
+  }
+
+  return base::android::ConvertUTF8ToJavaString(env, sync_code_hex);
+}
+
+base::android::ScopedJavaLocalRef<jstring>
+JNI_BraveSyncWorker_GetWordsFromSeedHex(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jcaller,
+    const base::android::JavaParamRef<jstring>& seed_hex) {
+  std::string str_seed_hex = base::android::ConvertJavaStringToUTF8(seed_hex);
+  DCHECK(!str_seed_hex.empty());
+
+  std::vector<uint8_t> bytes;
+  std::string sync_code_words;
+  if (base::HexStringToBytes(str_seed_hex, &bytes)) {
+    DCHECK_EQ(bytes.size(), SEED_BYTES_COUNT);
+    if (bytes.size(), SEED_BYTES_COUNT) {
+      sync_code_words = brave_sync::crypto::PassphraseFromBytes32(bytes);
+      std::vector<std::string> splitted_code_words = base::SplitString(
+          sync_code_words, " ", base::WhitespaceHandling::TRIM_WHITESPACE,
+          base::SplitResult::SPLIT_WANT_NONEMPTY);
+      if (splitted_code_words.size() != SEED_BIP39_WORD_COUNT) {
```

Comment:
This check is redundant. We only need to check if `sync_code_words` is empty here

Reply:
Fixed at 98fc8e626df92308ef293d4b3e3a477aa7e485b6